### PR TITLE
Remove usage of deprecated methods in elasticsearch client

### PR DESCRIPTION
**Pull request by marcinsbd**
## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/AuthRestClientConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+{
+    private final Optional<AwsSecurityConfig> awsSecurityConfig;
+    private final Optional<PasswordConfig> passwordConfig;
+
+    @Inject
+    AuthRestClientConfigurator(Optional<AwsSecurityConfig> awsSecurityConfig, Optional<PasswordConfig> passwordConfig)
+    {
+        this.awsSecurityConfig = awsSecurityConfig;
+        this.passwordConfig = passwordConfig;
+    }
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/AuthRestClientConfigurator.java:44]: Is it possible to split it further - Having an implementation specific to `Password` based authentication and an implementation specific to `AWS` based implementation.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java [hunk 2]
```diff
@@ -91,7 +120,7 @@ public void close()
     private static boolean isBackpressure(Throwable throwable)
     {
         return (throwable instanceof ResponseException) &&
-                (((ResponseException) throwable).getResponse().getStatusLine().getStatusCode() == RestStatus.TOO_MANY_REQUESTS.getStatus());
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java:94]: Can we remove it ?

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java [hunk 3]
```diff
@@ -67,19 +72,43 @@ public BackpressureRestClient(RestClient delegate, ElasticsearchConfig config, T
 
     public void setHosts(HttpHost... hosts)
     {
-        delegate.setHosts(hosts);
+        delegate.setNodes(stream(hosts).map(Node::new).collect(toImmutableList()));
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java:75]: Can we have something like this 
```
        delegate.setNodes(Stream.of(hosts)
                .map(Node::new)
                .collect(toImmutableList()));
```

Post `Stream` - we could move each operation to a new line

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java [hunk 4]
```diff
@@ -67,19 +72,43 @@ public BackpressureRestClient(RestClient delegate, ElasticsearchConfig config, T
...
     }
 
     public Response performRequest(String method, String endpoint, Map<String, String> params, HttpEntity entity, Header... headers)
             throws IOException
     {
-        return executeWithRetries(() -> delegate.performRequest(method, endpoint, params, entity, headers));
+        return executeWithRetries(() -> delegate.performRequest(toRequest(method, endpoint, params, entity, headers)));
+    }
+
+    private static Request toRequest(String method, String endpoint, Header... headers)
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java:90]: Can we move this method - below the `private static Request toRequest(String method, String endpoint, Map<String, String> params, HttpEntity entity, Header... headers)`

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java [hunk 5]
```diff
@@ -84,28 +84,46 @@ public void close()
...
+    public ClearScrollResponse clearScroll(ClearScrollRequest clearScrollRequest, RequestOptions requestOptions)
+            throws IOException
+    {
+        return executeWithRetries(() -> delegate.clearScroll(clearScrollRequest, requestOptions));
     }
 
     private static boolean isBackpressure(Throwable throwable)
     {
         return (throwable instanceof ElasticsearchStatusException) &&
-                (((ElasticsearchStatusException) throwable).status() == RestStatus.TOO_MANY_REQUESTS);
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java:108]: Can we revert this changes ?

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java [hunk 6]
```diff
@@ -84,28 +84,46 @@ public void close()
         delegate.close();
     }
 
-    public SearchResponse search(SearchRequest searchRequest, Header... headers)
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java:87]: Is it possible to split them into two methods - First we could remove the unused `Header...` and then we could remove the deprecated APIs. This makes it a bit difficult on identifying which is deprecated API. WDYT ?

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java [hunk 7]
```diff
@@ -84,28 +84,46 @@ public void close()
...
+    public SearchResponse search(SearchRequest searchRequest)
             throws IOException
     {
-        return executeWithRetries(() -> delegate.search(searchRequest, headers));
+        return search(searchRequest, RequestOptions.DEFAULT);
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java:90]: We can inline the method today right ?

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java [hunk 8]
```diff
@@ -230,45 +216,14 @@ private static BackpressureRestHighLevelClient createClient(
...
-            passwordConfig.ifPresent(securityConfig -> {
-                CredentialsProvider credentials = new BasicCredentialsProvider();
-                credentials.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(securityConfig.getUser(), securityConfig.getPassword()));
-                clientBuilder.setDefaultCredentialsProvider(credentials);
-            });
-
-            awsSecurityConfig.ifPresent(securityConfig -> clientBuilder.addInterceptorLast(new AwsRequestSigner(
-                    securityConfig.getRegion(),
-                    getAwsCredentialsProvider(securityConfig))));
+            configurators.forEach(c -> c.configure(clientBuilder));
```
Comment (Praveen2112) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java:219]: nit: configurator

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java [hunk 9]
```diff
@@ -67,19 +72,43 @@ public BackpressureRestClient(RestClient delegate, ElasticsearchConfig config, T
...
+    {
+        requireNonNull(headers, "headers cannot be null");
+        Request request = new Request(method, endpoint);
+        final RequestOptions.Builder options = request.getOptions().toBuilder();
```
Comment (martint) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java:94]: Don't use `final` for local variables.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java [hunk 10]
```diff
@@ -67,19 +72,43 @@ public BackpressureRestClient(RestClient delegate, ElasticsearchConfig config, T
...
+        for (Header header : headers) {
+            requireNonNull(header, "header cannot be null");
```
Comment (martint) [inline plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestClient.java:96]: This is unnecessary. The code will fail with NPE in the line below anyway if `header` is null.

File: plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java (hunk 6 above)
Comment (marcinsbd) [reply plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/BackpressureRestHighLevelClient.java:87]: You meant to commits, right? 
Done

**Review by kokosing — APPROVED** [review]
